modules/mobile/app/helpers/mobile/appointments_cache_interface.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module Mobile
+  class AppointmentsCacheInterface
+    def initialize
+      @now = DateTime.now.utc
+    end
+
+    def fetch_appointments(user:, start_date: nil, end_date: nil, fetch_cache: true)
+      appointments = nil
+      search_start_date = [start_date, latest_allowable_cache_start_date].compact.min
```

Comment:
If no start or end date is given, it compares the `latest_allowable_cache_start_date` with `latest_allowable_cache_start_date.iso8601` (same with soonest). Don't think this actually changes behavior but can be confusing. could we pull the .iso8601 in with the `latest_allowable_cache_start_date` and `soonest_allowable_cache_end_date` methods?

Reply:
Not exactly, because iso8601 is a string format, which breaks the `.min`/`.max` functionality. I did consider having different methods for "as date" and "as string" but struggled to figure out naming conventions that weren't ugly. Can you think of a good naming convention for that?

Reply:
could we go the other direction and only use `iso8601` right before we give it to the upstream service? that way that min/max comparison is always apples to apples (date object to date object).

Reply:
So you prompted me to take another look at this. Refactoring the dates into the new service object was one of the more painful parts of this because I kept hitting errors and didn't understand why. It turns out the problem was that `DateTime.now.utc` is a datetime, but calling `beginning_of_year - 1.year` converts it to a time. But the params contract in the controller expects a datetime. Calling iso8601 on it converts it to a string but the contract is evidently smart enough to recognize that as a datetime but will fail if you give it a time. The other issues I encountered with date comparisons were probably just me screwing something up.

So I've changed the default start and end dates to datetimes and everything seems happy now. Thanks for pushing me to look more closely at this because I also thought those conversions were muddying the waters.